logging/src/test/kotlin/com/ritense/logging/service/LoggingEventServiceIT.kt

```diff
@@ -0,0 +1,123 @@
+import com.ritense.logging.domain.LoggingEventPropertyId
+import com.ritense.logging.web.rest.dto.LoggingEventPropertyDto
+import com.ritense.logging.web.rest.dto.LoggingEventSearchRequest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Pageable
+import org.springframework.data.domain.Sort
+import org.springframework.transaction.annotation.Transactional
+import java.time.Duration
+import java.time.LocalDateTime
+import java.time.ZoneId
+import kotlin.random.Random
+
+@Transactional
+class LoggingEventServiceIT : BaseIntegrationTest() {
+
+    @Test
+    fun `should search logs paginated`() {
+        val now = LocalDateTime.now()
+        saveLoggingEvent("event 1", now - Duration.ofDays(1))
+        val searchRequest = LoggingEventSearchRequest(beforeTimestamp = now - Duration.ofDays(2))
+        saveLoggingEvent("event 2", now - Duration.ofDays(3))
+        saveLoggingEvent("event 3", now - Duration.ofDays(4))
+        saveLoggingEvent("event 4", now - Duration.ofDays(5))
+        val sort = Sort.by(Sort.Direction.DESC, "timestamp")
+
+        val page1 = loggingEventService.searchLoggingEvents(searchRequest, PageRequest.of(0, 2).withSort(sort))
+        val page2 = loggingEventService.searchLoggingEvents(searchRequest, PageRequest.of(1, 2).withSort(sort))
+
+        assertEquals(2, page1.count())
+        assertEquals("event 2", page1.content[0].formattedMessage)
+        assertEquals("event 3", page1.content[1].formattedMessage)
+        assertEquals(1, page2.count())
+        assertEquals("event 4", page2.content[0].formattedMessage)
+    }
+
+    @Test
+    fun `should search logs by search queries`() {
+        val yesterday = LocalDateTime.now() - Duration.ofDays(1)
+        saveLoggingEvent("test event", yesterday - Duration.ofDays(1))
+        saveLoggingEvent("another event", yesterday - Duration.ofDays(1))
+        val searchRequest = LoggingEventSearchRequest(
+            afterTimestamp = yesterday.minusDays(2),
+            beforeTimestamp = yesterday,
+            level = "INFO",
+            likeFormattedMessage = "test",
+        )
+
+        val results = loggingEventService.searchLoggingEvents(searchRequest, Pageable.unpaged()).content
+
+        assertEquals(1, results.size)
+        assertEquals("test event", results[0].formattedMessage)
+    }
+
+    @Test
+    fun `should search logs by search properties`() {
+        val yesterday = LocalDateTime.now() - Duration.ofDays(1)
+        val event1 = saveLoggingEvent("test event", yesterday - Duration.ofDays(1))
+        saveLoggingEvent("another event", yesterday - Duration.ofDays(1))
+        loggingEventPropertyRepository.save(
+            LoggingEventProperty(
+                id = LoggingEventPropertyId(event1.id, "key"),
+                event = event1,
+                value = "value"
+            )
+        )
```

Comment:
I would like to add another property just to validate the the property being filtered on doesn't cause the other properties to not be loaded. (I think this happens correctly automatically)

Reply:
added